### PR TITLE
Modified the following code in the AsyncProcessingService

### PR DESCRIPTION
**Pull request by SandeepNayak077**
1) Added new variable "_workerThreadCount" to handle the worker thread and to resolve OnStop event error, because thread pool may be shared by other tasks, such as .NET runtime
2) Did code modification related to log information, for debugging purpose

### CODE REVIEW

File: root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs [hunk 1]
```diff
@@ -307,6 +316,9 @@ private void WorkerThreadCallBack(object asyncTask)
         {
             AsyncProcessingServiceReturnValue selectedAsyncTask = (AsyncProcessingServiceReturnValue)asyncTask;
 
+            // A new worker thread started an Async task
+            this._workerThreadCount++;
```
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs]: good

File: root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs [hunk 2]
```diff
@@ -354,11 +368,24 @@ private void WorkerThreadCallBack(object asyncTask)
...
-                AsyncProcessingServiceReturnValue asyncReturnValue = this.UpdateAsyncTask(selectedAsyncTask, AsyncTaskUpdate.FAIL);
-                LogIF.ErrorLog("ASYNC-SERVICE", string.Format(GetMessage.GetMessageDescription("E0006"), selectedAsyncTask.TaskId, ex.Message.ToString()));
+                string exceptionInfo = "ErrorMessageID: " + ex.GetType().Name + Environment.NewLine + "ErrorMessage: " + ex.Message;
+                this.UpdateAsyncTask(selectedAsyncTask, AsyncTaskUpdate.FAIL, exceptionInfo);
+                LogIF.ErrorLog("ASYNC-SERVICE", string.Format(GetMessage.GetMessageDescription("E0006"), selectedAsyncTask.TaskId, ex.Message));
+            }
+            finally
+            {
+                // Async task is over
+                this._workerThreadCount--;
```
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs]: good

File: root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs [hunk 3]
```diff
@@ -516,21 +553,17 @@ public void StopAsyncProcess()
...
-            int freeWorkerThreads = 0;
-            int completionPortThreads = 0;
-
-            // Get available threads.
-            ThreadPool.GetAvailableThreads(out freeWorkerThreads, out completionPortThreads);
+            // Wait for all worker thread to be complete
+            LogIF.InfoLog("ASYNC-SERVICE", GetMessage.GetMessageDescription("I0004"));
 
-            while (freeWorkerThreads != this._maxThreadCount)
+            while (this._workerThreadCount != 0)
```
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs]: good

**Comment by SandeepNayak077** [comment]
Thank you for the compliment Nishino-san

**Pull request merged**